fs/dafilesrv/dafilesrv.cpp

```diff
@@ -386,7 +404,17 @@ int main(int argc, const char* argv[])
     StringBuffer componentName;
 
     // NB: bare-metal dafilesrv does not have a component specific xml
-    Owned<IPropertyTree> config = loadConfiguration(defaultYaml, argv, "dafilesrv", "DAFILESRV", nullptr, nullptr);
+    Owned<IPropertyTree> extractedGlobalConfig = createPTree("dafilesrv");
+
+#ifndef _CONTAINERIZED
+    Owned<IPropertyTree> env = getHPCCEnvironment();
+    IPropertyTree* globalTracing = env->queryPropTree("Software/tracing");
+    if (globalTracing != nullptr)
+        extractedGlobalConfig->addPropTree("tracing", globalTracing);
+#endif
+
+    // NB: bare-metal dafilesrv does not have a component specific xml, extracting relevant global configuration instead
+    Owned<IPropertyTree> config = loadConfigurationWithGlobalDefault(defaultYaml, extractedGlobalConfig, argv, "dafilesrv", "DAFILESRV");
```

Comment:
instead of new function/adding to global, could you add 'tracing' to the component config instead?
e.g.:

```
#ifndef _CONTAINERIZED
    Owned<IPropertyTree> env = getHPCCEnvironment();
    IPropertyTree* tracing = env->getPropTree("Software/tracing");
    if (tracing)
        config->setPropTree("tracing", tracing);
#endif
```

(and combine with #else // __CONTAINERIZED block below)